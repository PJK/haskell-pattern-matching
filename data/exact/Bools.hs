and' :: Bool -> Bool -> Bool
and' True True = True
and' False _ = False

-- Evaluateness should be:
--
-- Evaluatedness
--  "and'"
--  [ ( [ WildcardPattern, WildcardPattern]
--    , [ [ConstructorPattern "True" [], ConstructorPattern "False" []]
--      , []
--      ]
--    )
--  , ( [ConstructorPattern "True", WildcardPattern]
--    , [ [ConstructorPattern "True" []]
--      , [ConstructorPattern "True [], ConstructorPattern "False" []]
--      ]
--  ]
--
--
--  The constraints are:
--  For the first clause:
--   and' x    y where x is bottom will diverge
--   and' True y where y is bottom will diverge
--  For the second clause: nothing
--
--
