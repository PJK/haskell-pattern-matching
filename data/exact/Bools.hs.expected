{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "and'",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "False",
                                []
                            ]
                        }
                    ]
                ]
            }
        ]
    ]
}
